core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
+                  ,'address','country','city','state','zip'
+                  ,'comment','extended'
+                )
+               ,'name'         => 'username'
+               ,'description'  => 'email'
+               ,'content'      => 'extended'
+               ,'icon'         => 'user'
+            );
+            /*  system settings :x:ss:site_start: 
+                2do: Doesn't really make sense before we can open system settings for a specific key...
+            $elements['ss'] = array(
+                 'label'       => '(ss) systemSetting'
+                ,'type'        => ''
+                ,'class_key'   => 'modSystemSetting'
+                ,'permission'  => $this->modx->hasPermission('settings')
+                ,'action'      => 'system/settings'
+                ,'fields'      => array('key')
+                ,'name'        => 'key'
+                ,'description' => 'value'
+                ,'content'     => 'value'
+                ,'icon'        => 'gear'
+                
+            );
+            */
+            
+        }     
+        
+        
+        
+
+    /* Basic UI: Messages and validation */
+        $valid = false;
+        $step1 = $step2 = $step3 = $step4 = $step5 = false;
+    
+        /* What are we looking for? */
+        $start      = isset($query[0]) AND  empty($query) ? true : false;
+        $element    = isset($query[1]) ? $query[1] : false;
+        $field      = isset($query[2]) ? $query[2] : false;
+        $value      = isset($query[3]) ? $query[3] : false;
+        $page       = isset($query[4]) ? $query[4] : false;
+        
+        $count = count($query);
+        $result['count'] = $count;
+
+        $btn = "x-btn x-btn-small x-btn-icon-small-left primary-button x-btn-noicon";
+        
+        function validate_Uberbar($elements,$query){
+            
+            global $modx;
+            
+            $uberbar_mode = $modx->getOption('uberbar_mode');
+            
+            $count = count($query);
+
+            $element    = isset($query[1]) ? $query[1] : false;
+            $field      = isset($query[2]) ? $query[2] : false;
+            $value      = isset($query[3]) ? $query[3] : false;
+            $page       = isset($query[4]) ? $query[4] : false;
+            
+            // Element related stuff...
+            if($count > 2)
+            {
+                /* Validate elements */
+                if( $element != '*' && !array_key_exists($element,$elements) )
+                {
+                    // Error: Unknown element
+                        $result['header'] = $modx->lexicon('uberbar_error_onknown_element_header');
+                        
+                    // You are searching for an unknown element: <strong>{element}</strong>;
+                        $s = array('{element}');
+                        $r = array($element);
+                        $result['msg'] = str_replace( $s , $r, $modx->lexicon('uberbar_error_onknown_element_msg') );
```

Comment:
As long as you use your xray vision and not your supoer strong microwaves, I don't mind. I've learned so many things in the last minute, that I can only say. Thank you Sir!